src/Mbstring/Mbstring.php

```diff
@@ -991,4 +995,66 @@ private static function getEncoding($encoding)
 
         return $encoding;
     }
+
+    public static function mb_trim(string $string, ?string $characters = null, ?string $encoding = null): string
+    {
+        return self::mb_internal_trim('^[%s]+|[%s]+$', $string, $characters, $encoding);
+    }
+
+    public static function mb_ltrim(string $string, ?string $characters = null, ?string $encoding = null): string
+    {
+        return self::mb_internal_trim('^[%s]+', $string, $characters, $encoding);
+    }
+
+    public static function mb_rtrim(string $string, ?string $characters = null, ?string $encoding = null): string
+    {
+        return self::mb_internal_trim('[%s]+$', $string, $characters, $encoding);
+    }
+
+    private static function mb_internal_trim(string $regex, string $string, ?string $characters = null, ?string $encoding = null): string
+    {
+        if (null === $encoding) {
+            $encoding = mb_internal_encoding();
+        }
+
+        try {
+            $validEncoding = @mb_check_encoding('', $encoding);
+        } catch (\ValueError $e) {
+            throw new \ValueError(sprintf('%s(): Argument #3 ($encoding) must be a valid encoding, "%s" given.', debug_backtrace()[1]['function'], $encoding));
+        }
+
+        // BC for PHP 7.3 and lower
+        if (!$validEncoding) {
+            throw new \ValueError(sprintf('%s(): Argument #3 ($encoding) must be a valid encoding, "%s" given.', debug_backtrace()[1]['function'], $encoding));
+        }
+
+        if ('' === $characters) {
+            return null === $encoding ? $string : mb_convert_encoding($string, $encoding);
+        }
+
+        if (null === $characters) {
+            $characters = self::CHARACTERS;
+        }
+
+        $regexCharacter = preg_quote($characters ?? '', '/');
+        $regex = sprintf($regex, $regexCharacter, $regexCharacter);
+
+        if ('ASCII' === mb_detect_encoding($characters) && 'ASCII' === mb_detect_encoding($string) && !empty(array_intersect(str_split(self::CHARACTERS), str_split($string)))) {
+            $options = 'g';
+        } else {
+            $options = '';
+        }
+        
+        try {
+            $test = mb_ereg_replace($regex, "", $string, $options);
+
+            if (null === $test) {
+                throw new \Exception();
+            }
+
+            return $test;
+        } catch (\Exception $e) {
+            return preg_replace(sprintf('/%s/', $regex), "", $string);
```

Comment:
let's use concatenation for such simple things (or better: add the `/` from the start if we know we're always going to use PCRE)